File: Ad-Hoc/2231/2231.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int n,m, teste = 1, maior = INT32_MIN, menor = INT32_MAX, aux = 0;
    cin >> n >> m;
    while (n!=0 && m!=0){
        int *vet = (int*) malloc(sizeof(int) * n);
        for (int i = 0; i<n; i++) cin >> vet[i];
        for (int i = 0; i<=n-m; i++){
            aux = 0;
            for (int j = i; j<i+m; j++) aux += vet[j];
            maior = max(maior, aux/m);
            menor = min(menor, aux/m);
        }
        cout << "Teste " << teste << endl << menor << " " << maior << endl << endl;
        teste++;
        cin >> n >> m;
        free(vet);
        maior = INT32_MIN;
        menor = INT32_MAX;
    }
    return 0;
}

File: Ad-Hoc/2231/2231.py
N,M = map(int, input().split())
count = 1
while N>=0 and N<=10000 and M>=1 and M<=N:
    lista = []
    aux = []
    soma = 0
    condition = 0
    for x in range(N):
        z = int(input())
        aux.append(z)
        soma += z
        condition+=1
        if condition == M:
            lista.append(soma)
            soma -= aux[0]
            condition-=1
            aux.pop(0)
    print(f"Teste {count}")
    print(f"{int(min(lista)/M)} {int(max(lista)/M)}")
    print()
    count+=1
    N, M = map(int, input().split())